src/SmartTransactionsController.ts

```diff
@@ -889,4 +889,56 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
       );
     });
   }
+
+  #wipeSmartTransactionsPerChainId(
+    chainId: Hex,
+    selectedAddressLowerCase: string,
+  ): void {
+    const { smartTransactions } = this.state.smartTransactionsState;
+    const smartTransactionsForSelectedChain: SmartTransaction[] =
+      smartTransactions?.[chainId];
+    if (
+      !smartTransactionsForSelectedChain ||
+      smartTransactionsForSelectedChain.length === 0
+    ) {
+      return;
+    }
+    const newSmartTransactionsForSelectedChain =
+      smartTransactionsForSelectedChain.filter(
+        (smartTransaction: SmartTransaction) =>
+          smartTransaction.txParams?.from !== selectedAddressLowerCase,
+      );
+    this.update({
+      smartTransactionsState: {
+        ...this.state.smartTransactionsState,
+        smartTransactions: {
+          ...smartTransactions,
+          [chainId]: newSmartTransactionsForSelectedChain,
+        },
+      },
+    });
+  }
+
+  wipeSmartTransactions(
+    selectedAddress?: string,
+    ignoreNetwork?: boolean,
+  ): void {
+    if (!selectedAddress) {
+      return;
+    }
+    const selectedAddressLowerCase = selectedAddress.toLowerCase();
+    if (ignoreNetwork) {
+      this.#wipeSmartTransactionsPerChainId(
+        this.config.chainId,
+        selectedAddressLowerCase,
+      );
+    } else {
+      this.config.supportedChainIds.forEach((chainId) => {
```

Comment:
Rather than looping through every supported chain ID, should we instead iterate over the state itself and the chain IDs in it?

Could this way introduce gaps if the config is changed to support a smaller set of chain IDs and old transactions exist with chain IDs no longer supported?